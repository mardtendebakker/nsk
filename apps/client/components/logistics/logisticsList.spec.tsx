import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import LogisticsList from './logisticsList';

function RenderC({ onClick } : { onClick:(logistiId: number) => void }) {
  return (
    <LogisticsList
      logistics={[{
        id: 1,
        username: 'username',
        firstname: 'firstname',
        lastname: 'lastname',
        email: 'email@gmail.com',
      }, {
        id: 2,
        username: 'username2',
        firstname: 'firstname2',
        lastname: 'lastname2',
        email: 'email2@gmail.com',
      }]}
      onClick={onClick}
      selectedLogisticIds={[1]}
    />
  );
}

describe('LogisticsList component', () => {
  it('matches snapshot', () => {
    const { asFragment, getByText } = render(<RenderC onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText('username')).toBeInTheDocument();
    expect(getByText('username2')).toBeInTheDocument();
  });
  it('handles click properly', () => {
    const onClick = jest.fn();
    const { getByText } = render(<RenderC onClick={onClick} />);
    fireEvent.click(getByText('everyone'));
    fireEvent.click(getByText('username'));
    fireEvent.click(getByText('username2'));

    expect(onClick).toHaveBeenCalledWith(0);
    expect(onClick).toHaveBeenCalledWith(1);
    expect(onClick).toHaveBeenCalledWith(2);
  });
});
