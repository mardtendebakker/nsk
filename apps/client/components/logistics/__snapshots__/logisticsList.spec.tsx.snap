// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    role="menu"
    tabindex="-1"
  >
    <li
      class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-165f8jw-MuiButtonBase-root-MuiMenuItem-root"
      role="menuitem"
      tabindex="0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
      >
        everyone
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </li>
    <li
      class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-17nos47-MuiButtonBase-root-MuiMenuItem-root"
      role="menuitem"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1we0tu4-MuiTypography-root"
      >
        username
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </li>
    <li
      class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-5785nc-MuiButtonBase-root-MuiMenuItem-root"
      role="menuitem"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
      >
        username2
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </li>
  </ul>
</DocumentFragment>
`;
